### ISSUE TITLE
SSIM loss fail when batchsize>1

### ISSUE THREAD

**Issue by Can-Zhao** [Project-MONAI/MONAI#5886]
**Describe the bug**
SSIM loss fail when batchsize>1

**To Reproduce**
```
import torch
from monai.losses.ssim_loss import SSIMLoss
x = torch.ones([4,1,10,10])/2
y = torch.ones([4,1,10,10])/2
data_range = x.max().unsqueeze(0)
print(1-SSIMLoss(spatial_dims=2)(x,y,data_range))
```

**Expected behavior**
Should print 1, but got error

**Screenshots**

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'
```

**Additional context**

**Comment by Can-Zhao** [comment]
Hi @PedroFerreiradaCosta I was wondering if you would have time to look at it. If not, I could work on it in two weeks. Thank you a lot!

**Comment by jak0bw** [comment]
The same error occurred for me. After some investigation I think I found the problem. 

The problem lies in https://github.com/Project-MONAI/MONAI/blob/8a5a55093aebdf631b2fec14b2111a3c980ef584/monai/losses/ssim_loss.py

If the batchsize is >1, SSIMLoss calls SSIMMetric for each batch item individually . The results are collected in the ssim_value variable. Before the concatenation ssim_value and the new ssim value of the current batch item are given a new view of size 1. This call to view() raises an error at the second call as ssim_value has now an absolute size of 2 https://github.com/Project-MONAI/MONAI/blob/8a5a55093aebdf631b2fec14b2111a3c980ef584/monai/losses/ssim_loss.py#L98

An error only occurs for a batchsize >2 as before ssim_value has a size of 1 before concatenation. When the batchsize == 1 another path is executed, avoiding the concatenation and the view() call.

I opened a pull request #5908  fixing the view() call parameter for ssim_value from 1 to the loop variable i. i contains the current batch item index which fortunately always is the current size of ssim_value. Therefore, this issue is avoided.
